Removes IngressBandwidth Annotation

danehans opened the pull request:
The IngressBandwidth is unimplemented but is represented as an exportable constant. This PR removes the IngressBandwidth annotation to avoid potential confusion and misuse.

Fixes: #25532

jschwinger233 reviewed (commented):
Could you please explain why we intend to remove this annotation from the package constant?  I understand the reasons mentioned in the Slack thread, but it would be preferable to have them documented in the commit message.

borkmann commented on daemon/cmd/endpoint.go:
```diff
@@ -430,12 +430,13 @@ func (d *Daemon) createEndpoint(ctx context.Context, owner regeneration.Owner, e
 			addLabels.MergeLabels(identityLabels)
 			infoLabels.MergeLabels(info)
-			if _, ok := annotations[bandwidth.IngressBandwidth]; ok {
+			ingBandwidth := "kubernetes.io/ingress-bandwidth"
```
How is this any better? I thought from your explanation that bandwidth.IngressBandwidth is unused and should be removed. Yet it is used in the code. So your change makes it worse.

danehans replied:
@borkmann thanks for the review. IMHO this is better b/c the annotation can no longer be imported by another pkg ([xref](https://go.dev/ref/spec#Exported_identifiers)). @pchaigno suggested in the associated Slack thread to remove the constant. Based on your feedback, I will update the PR to remove the associated log message.

pchaigno replied:
Did you consider simply not exporting the constant outside of the package where it's defined, eventually by moving it to another package?

danehans replied:
@pchaigno ^ is an option too. Does the team prefer the removal of the constant and associated log message or move the constant to the cmd pkg, make it internal, and keep the log message?

danehans committed:
Unexports IngressBandwidth Annotation

IngressBandwidth is unimplemented but is represented as an exportable
constant. This PR moves the IngressBandwidth constant to the cmd pkg
as local declaration to avoid potential confusion and misuse.

Signed-off-by: Daneyon Hansen <daneyon.hansen@solo.io>

brb reviewed (changes requested):
Thanks for the PR!

However, we tend to keep all related consts in the same place, and [our docs](https://docs.cilium.io/en/stable/network/kubernetes/bandwidth-manager/) already should protect from a potential misuse.
